pkg/apis/provisioning/v1alpha5/labels.go

```diff
@@ -103,7 +104,13 @@ func IsRestrictedLabel(key string) error {
 // They are either known labels that will be injected by cloud providers,
 // or label domain managed by other software (e.g., kops.k8s.io managed by kOps).
 func IsRestrictedNodeLabel(key string) bool {
```

Comment:
I'm struggling to understand how this method is different from `IsRestrictedLabel`

Reply:
The underneath logic for these two methods are different. The `IsRestrictedLabel` is filtering labels that are allowed to be used in Karpenter provisioner. `IsRestrictedNodeLabel` is filtering which labels should be injected by Karpenter when the node object is created.

For example, the well-known labels are allowed by `IsRestrictedLabel` but `IsRestrictedNodeLabel` will reject because those labels should be injected by cloudprovider.